qa/test_import.py

```diff
@@ -0,0 +1,37 @@
+# try to import all .py files from a given directory
+
+import argparse
+import glob
+import os
+import importlib
+import importlib.util
+
+
+def _module_name(path):
+    return os.path.splitext(os.path.basename(path))[0]
+
+
+def _import_file(path):
+    mod_name = _module_name(path)
+    print('Importing module {} from {}'.format(mod_name, path))
+    mod_spec = importlib.util.spec_from_file_location(mod_name, path)
+    mod = importlib.util.module_from_spec(mod_spec)
+    mod_spec.loader.exec_module(mod)
+    print('Imported module {} from {} successfully'.format(mod_name, path))
+    return mod_name
+
+
+def _parser():
+    parser = argparse.ArgumentParser(
+        description='Try to import a file',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('path', nargs='+', help='Glob to select files')
+    return parser
```

Comment:
```suggestion
def _module_name(path):
    task = os.path.splitext(path)[0]
    parts = task.split(os.path.sep)
    package = parts[0]
    name = ''.join('.' + c for c in parts[1:])
    return package, name

def _import_file(path):
    package, mod_name = _module_name(path)
    line = f'Importing {package}{mod_name} from {path}'
    print(f'{line:<80}', end='')
    mod_spec = importlib.util.find_spec(mod_name, package)
    mod = mod_spec.loader.load_module()
    if mod is None:
        result = 'FAIL'
    else:
        result = 'DONE'
    print(f'{result:>6}')
    mod_spec.loader.exec_module(mod)

def _parser():
    parser = argparse.ArgumentParser(
        description='Try to import a file',
        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    parser.add_argument('path', nargs='+', help='Glob to select files')
    return parser
```

Reply:
@toabctl hi Thomas, i rewrote the code to load the module. as without this change, the test is not able to load submodule correctly.

i think that's why we have the following-up fixes of #34173 and #34195.